### ISSUE TITLE
Windows dual monitor playback glitches

### ISSUE THREAD

**Issue by chowey** [mpv-player/mpv#6463]
### mpv version and platform

I am using mpv 0.29.1 32-bit on Windows (lachs0r build).

### Reproduction steps

I have two monitors from different manufacturers with different resolutions. Both are 60Hz. I am using `profile=gpu-hq`. I have a video that is 60 FPS (dropframe).

I play the video from mpv player, fullscreen, on the secondary monitor (not the primary monitor) with `video-sync=display-resample`. The video looks glitchy. With the info overlay, I see that the the "Display FPS (estimated)" is about 59.8. There are mis-timings and dropped frames.

I disable my primary monitor, so that the secondary monitor becomes the primary monitor. I play the same video from mpv player in fullscreen mode. The video looks smooth. With the info overlay, I see that the "Display FPS (estimated)" is about 60.0. There are almost no mis-timings or dropped frames.

It appears that in a dual-monitor setup, mpv player is estimating the Display FPS using the wrong monitor. This messes up its timing and the playback looks glitchy.

### Expected behavior

mpv player should use the Display FPS (estimated) from the monitor that the video is playing on, not from the primary monitor.

### Actual behavior

It appears that mpv player is actually using the primary monitor to calculate Display FPS (estimated). The video playback looks glitchy every so often.

I realize I am giving my conclusions as well as my observations, but there is definitely something wrong when I play 60 FPS video on a secondary monitor on Windows.

**Comment by EzeKees** [comment]
What models of video cards do you have?

**Comment by chowey** [comment]
It looks like NVIDIA Quadro K620.

**Comment by EzeKees** [comment]
Thank you. I was asking you why I have a similar problem (I have a nVIDIA GEFORCE GT 610 and an Intel HD 4600) but the problem is ineffective to mpv. It happens to me with any application in full screen and in Windows 10

**Comment by chowey** [comment]
Yeah my issue happens only when there are two monitors and mpv player is on the second monitor.

**Comment by chowey** [comment]
I recompiled mpv player with #6329 and this fixes my glitches when I run in exclusive fullscreen mode. So this is a workaround for others experiencing the same issue.

**Comment by chowey** [comment]
Granted... but until mpv player has accepted the patch, is this issue really resolved?

Also, does anyone else have the problem with exclusive fullscreen where it causes both your displays to flicker black when you start it?

**Comment by lachs0r** [comment]
Yes, that flickering is also a flaw in Windows’ compositor and possibly drivers.

**Comment by chowey** [comment]
I didn't notice any improvement with `gpu-context=angle`, and I am working with Windows 7 boxes so they don't support Vulkan. Are you certain OpenGL will make Windows switch to exclusive fullscreen mode?

**Comment by lachs0r** [comment]
We used to get bug reports with users complaining about flickering and taskbar issues when switching to fullscreen with OpenGL, so I’m quite sure it does.

**Comment by chowey** [comment]
I have further found that this only occurs on Windows 7 and not on Windows 10. Maybe this is a difference between DirectX 11 and DirectX 12?

**Comment by jeeb** [comment]
If I understand it correctly, the better behavior is with the newer OS?

It's very possible that newer versions of Windows handle this better. Or if the hardware is different, it could also be down to the drivers (I have many not-so-great memories of Direct* behavior being affected by hardware drivers).

**Comment by chowey** [comment]
You understand correctly. The better behavior is with the newer OS.

There are other differences. The Windows 7 computer is using an older Nvidia K620, and the Windows 10 computer is using a newer Nvidia P620.

The Windows 7 computer has one monitor on DVI and the other monitor on Display Port. The Windows 10 computer has both monitors on Display Port.

I tried the same monitors on both computers, so that is not the difference.

**Comment by chowey** [comment]
Disabling the windows compositor on Windows 7 also fixes this issue. (It just creates other problems in other software.)

**Comment by chowey** [comment]
For now I have solved this by [disabling composition](https://docs.microsoft.com/en-us/windows/desktop/api/Dwmapi/nf-dwmapi-dwmenablecomposition) when I start MPV player:

```
DwmEnableComposition(DWM_EC_DISABLECOMPOSITION)
```

This is purely a hack and I'd love if someone else has a better idea.

**Comment by lachs0r** [comment]
The compositor has indeed improved with Windows 8+ (where it cannot be disabled anymore), however I’m not sure if it solves the issue of only running at a fixed refresh rate across all displays.
